### ISSUE TITLE
threads::shared: Synopsis code does not compile cleanly

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#5603]
Migrated from [rt.perl.org#9749](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=9749) (status was 'resolved')

Searchable as RT9749$

**Comment by p5pRT** [comment]
### From perl58@dev1.ridas.co.uk

#### Created by scriptyrich@yahoo.co.uk



&nbsp;  use threads;
&nbsp;  use threads&#8203;::shared;

&nbsp;  my $var : shared;

&nbsp;  my\($scalar\, @&#8203;array\, %hash\);
&nbsp;  share\($scalar\);
&nbsp;  share\(@&#8203;array\);
&nbsp;  share\(%hash\);
&nbsp;  my $bar = share\(\[\]\);
&nbsp;  $hash\{bar\} = share\(\{\}\);

Gives&#8203;:

&nbsp;  Type of arg 1 to threads&#8203;::shared&#8203;::share must be one of \[$@&#8203;%\] \(not single ref constructor\) at a\.pl line 12\, near "\]\)"
&nbsp;  Type of arg 1 to threads&#8203;::shared&#8203;::share must be one of \[$@&#8203;%\] \(not single ref constructor\) at a\.pl line 13\, near "\}\)"
&nbsp;  Execution of a\.pl aborted due to compilation errors\.

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=medium

Site configuration information for perl v5.8.0:

Configured by root at Sun Jun 16 00:35:43 BST 2002.

Summary of my perl5 (revision 5.0 version 8 subversion 0) configuration:
  Platform:
    osname=linux, osvers=2.4.18-6mdk, archname=i686-linux-thread-multi
    uname='linux dev1.ridas.co.uk 2.4.18-6mdk #1 fri mar 15 02:59:08 cet 2002 i686 unknown '
    config_args=''
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=define use5005threads=undef useithreads=define usemultiplicity=define
    useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm',
    optimize='-O2',
    cppflags='-D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -I/usr/local/include -I/usr/include/gdbm'
    ccversion='', gccversion='2.96 20000731 (Mandrake Linux 8.2 2.96-0.76mdk)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib
    libs=-lnsl -lgdbm -ldb -ldl -lm -lpthread -lc -lcrypt -lutil
    perllibs=-lnsl -ldl -lm -lpthread -lc -lcrypt -lutil
    libc=/lib/libc-2.2.4.so, so=so, useshrplib=false, libperl=libperl.a
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic'
    cccdlflags='-fpic', lddlflags='-shared -L/usr/local/lib'

Locally applied patches:
    


@INC for perl v5.8.0:
    /usr/local/perl/devel/5.8/lib/5.8.0/i686-linux-thread-multi
    /usr/local/perl/devel/5.8/lib/5.8.0
    /usr/local/perl/devel/5.8/lib/site_perl/5.8.0/i686-linux-thread-multi
    /usr/local/perl/devel/5.8/lib/site_perl/5.8.0
    /usr/local/perl/devel/5.8/lib/site_perl
    .


Environment for perl v5.8.0:
    HOME=/home/perl58
    LANG=en_GB
    LANGUAGE=en_GB:en
    LC_COLLATE=en_GB
    LC_CTYPE=en_GB
    LC_MESSAGES=en_GB
    LC_MONETARY=en_GB
    LC_NUMERIC=en_GB
    LC_TIME=en_GB
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/usr/local/perl/devel/5.8/bin:/bin:/usr/bin:/usr/X11R6/bin:/usr/local/bin:/usr/games:/usr/local/java/blackdown/j2sdk1.3.1/bin:/home/rich/bin
    PERL_BADLANG (unset)
    SHELL=/bin/bash


```
</details>

**Comment by p5pRT** [comment]
### From arthur@contiller.se

Hi\,

This has been fixed as of patchnumber \#17453\, the synopsis has changed
to the &share\(\[\]\) syntax instead of plain share\(\[\]\)\.

Arthur

**Comment by p5pRT** [comment]
arthur@contiller.se - Status changed from 'open' to 'resolved'
